toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt

```diff
@@ -126,42 +132,97 @@ public class TraceState(
         _selectedTraceConfiguration.value = config
     }
 
+    private var _currentTraceRun: TraceRun? = null
+    internal val currentTraceRun: TraceRun
+        get() = _currentTraceRun ?: throw IllegalStateException("TraceRun cannot be null")
+
+    private val currentTraceGraphics : MutableList<Graphic> = mutableListOf()
+
     /**
-     * TBD
+     * Run a trace on the Utility Network using the selected trace configuration and starting points.
+     *
+     * @return true if the trace results are available, false otherwise.
+     * @since 200.6.0
      */
-    public suspend fun trace() {
+    public suspend fun trace() : Boolean {
         // Run a trace
-        val utilityNetworkDefinition = utilityNetwork.definition
-        val utilityNetworkSource =
-            utilityNetworkDefinition!!.getNetworkSource("Electric Distribution Line")
-        val utilityAssetGroup = utilityNetworkSource!!.getAssetGroup("Medium Voltage")
-        val utilityAssetType =
-            utilityAssetGroup!!.getAssetType("Underground Three Phase")
-        val startingLocation = utilityNetwork.createElementOrNull(
-            utilityAssetType!!,
-            Guid("0B1F4188-79FD-4DED-87C9-9E3C3F13BA77")
-        )
+        val traceConfiguration = selectedTraceConfiguration.value ?: return false
 
-        val utilityTraceParameters = UtilityTraceParameters(
-            UtilityTraceType.Connected,
-            listOf(startingLocation!!)
-        )
+        if (startingPoints.isEmpty() && traceConfiguration.minimumStartingLocations == UtilityMinimumStartingLocations.One) {
+            // TODO: Handle error
+            return false
+        }
 
-        utilityNetwork.trace(
-            utilityTraceParameters
-        ).onSuccess {
+        if (startingPoints.size < 2 && traceConfiguration.minimumStartingLocations == UtilityMinimumStartingLocations.Many) {
+            // TODO: Handle error
+            return false
+        }
+
+        val utilityTraceParameters = UtilityTraceParameters(traceConfiguration, startingPoints.map { it.utilityElement })
+
+        var traceResults: List<UtilityTraceResult> = emptyList()
+
+        utilityNetwork.trace(utilityTraceParameters).onSuccess {
```

Comment:
It is more efficient to only allocate the empty list if there is an error.
```suggestion
val traceResults = utilityNetwork.trace(utilityTraceParameters).getOrElse { 
            //handle error
            println("ERROR")
            emptyList<UtilityElementTraceResult>()
        }
```

Reply:
updated